node/silkworm/stagedsync/prune_execution_test.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+   Copyright 2020-2021 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include <catch2/catch.hpp>
+#include <ethash/keccak.hpp>
+
+#include <silkworm/chain/config.hpp>
+#include <silkworm/chain/protocol_param.hpp>
+#include <silkworm/common/data_dir.hpp>
+#include <silkworm/common/temp_dir.hpp>
+#include <silkworm/db/buffer.hpp>
+#include <silkworm/db/stages.hpp>
+#include <silkworm/execution/address.hpp>
+#include <silkworm/execution/execution.hpp>
+#include <silkworm/rlp/encode.hpp>
+#include <silkworm/types/account.hpp>
+#include <silkworm/types/block.hpp>
+
+#include "stagedsync.hpp"
+
+TEST_CASE("Prune Execution without prune function") {
+    using namespace silkworm;
+
+    TemporaryDirectory tmp_dir;
+    DataDirectory data_dir{tmp_dir.path()};
+
+    // Initialize temporary Database
+    db::EnvConfig db_config{data_dir.get_chaindata_path().string(), /*create*/ true};
+    db_config.inmemory = true;
+    auto env{db::open_env(db_config)};
+    stagedsync::TransactionManager txn{env};
+    db::table::create_all(*txn);
+
+    // ---------------------------------------
+    // Prepare
+    // ---------------------------------------
+
+    uint64_t block_number{1};
+    auto miner{0x5a0b54d5dc17e0aadc383d2db43b0a0d3e029c4c_address};
+
+    Block block{};
+    block.header.number = block_number;
+    block.header.beneficiary = miner;
+    block.header.gas_limit = 100'000;
+    block.header.gas_used = 63'820;
+
+    // This contract initially sets its 0th storage to 0x2a
+    // and its 1st storage to 0x01c9.
+    // When called, it updates its 0th storage to the input provided.
+    Bytes contract_code{*from_hex("600035600055")};
+    Bytes deployment_code{*from_hex("602a6000556101c960015560068060166000396000f3") + contract_code};
+
+    block.transactions.resize(1);
+    block.transactions[0].data = deployment_code;
+    block.transactions[0].gas_limit = block.header.gas_limit;
+    block.transactions[0].max_priority_fee_per_gas = 0;  // EIP-1559
```

Comment:
Please change this test similar to my changes in PR #381. Either use `kLondonTestConfig` and leave EIP-1559 transactions (what I did in `unwind_execution_test`), or keep legacy transactions and equate `max_priority_fee_per_gas` with `max_fee_per_gas` (what I did in `history_index_test`).